code/modules/research/ordnance/scipaper_publisher.dm

```diff
@@ -0,0 +1,220 @@
+		singular_experiment["target"] = experiment.target_amount
+		if(istype(experiment, /datum/experiment/ordnance/explosive))
+			singular_experiment["suffix"] = "tiles"
+			singular_experiment["prefix"] = "Range"
+		else if(istype(experiment, /datum/experiment/ordnance/gaseous))
+			singular_experiment["suffix"] = "moles"
+			singular_experiment["prefix"] = "Gas"
+		parsed_experiments += list(singular_experiment)
+
+	var/list/parsed_partners = list()
+	for (var/datum/scientific_partner/partner in SSresearch.scientific_partners)
+		var/list/singular_partner = list()
+		singular_partner["name"] = partner.name
+		singular_partner["flufftext"] = partner.flufftext
+		singular_partner["multipliers"] = partner.multipliers
+		singular_partner["path"] = partner.type
+		singular_partner["boostedNodes"] = list()
+		singular_partner["acceptedExperiments"] = list()
+		for (var/node_id in partner.boosted_nodes)
+			var/datum/techweb_node/node = SSresearch.techweb_node_by_id(node_id)
+			singular_partner["boostedNodes"] += list(list("name" = node.display_name, "discount" = partner.boosted_nodes[node_id], "id"=node_id))
+		for (var/datum/experiment/ordnance/ordnance_experiment as anything in partner.accepted_experiments)
+			singular_partner["acceptedExperiments"] += initial(ordnance_experiment.name)
+		parsed_partners += list(singular_partner)
+
+	data["experimentInformation"] = parsed_experiments
+	data["partnersInformation"] = parsed_partners
+	data["coopIndex"] = SCIPAPER_COOPERATION_INDEX
+	data["fundingIndex"] = SCIPAPER_FUNDING_INDEX
+	return data
+
+/datum/computer_file/program/scipaper_program/ui_data()
+	// Program Headers:
+	var/list/data = get_header_data()
+	data["currentTab"]  = current_tab
+
+	// First page. Form submission.
+	if(current_tab == 1)
+		data["fileList"] = list()
+		data["expList"] = list()
+		data["allowedTiers"] = list()
+		data["allowedPartners"] =  list()
+		// Both the file and experiment list are assoc lists. ID as value, display name as keys.
+		for(var/datum/computer_file/data/ordnance/ordnance_file in holder.stored_files)
+			data["fileList"] += list(ordnance_file.filename = ordnance_file.uid)
+		if(selected_file)
+			for (var/possible_experiment in selected_file.possible_experiments)
+				var/datum/experiment/ordnance/experiment = possible_experiment
+				data["expList"] += list(initial(experiment.name) = experiment)
+		data["allowedTiers"] = paper_to_be.calculate_tier()
+		for (var/partner in SSresearch.scientific_partners)
+			var/datum/scientific_partner/scientific_partner = partner
+			if(paper_to_be.experiment_path in scientific_partner.accepted_experiments)
+				data["allowedPartners"] += list(scientific_partner.name = scientific_partner.type)
+
+		data += paper_to_be.return_gist()
+		data["selectedFile"] = selected_file?.filename
+		// Renamed both of these to be more readable.
+		data["selectedExperiment"] = data["experimentName"]
+		data -= "experimentName"
```

Comment:
This seems like a really strange pattern, why have them be unreadable in the first place?

Reply:
Figured the other two are more general, while selectedX conveys that it's a selection in the UI.

Reply:
So, do you still want me to get rid of it?